src/common/os_utils/src/os_utils_unix.cpp

```diff
@@ -0,0 +1,118 @@
+#include <os_utils_unix.hpp>
+
+#include <os_utils.hpp>
+
+#include <algorithm>
+#include <cerrno>
+#include <cstdio>
+#include <filesystem>
+#include <iostream>
+#include <sstream>
+#include <unistd.h>
+
+namespace
+{
+    struct PCloseDeleter
```

Comment:
You can use file smart deleter: https://github.com/wazuh/wazuh-agent/blob/main/src/common/deleterHelper/include/fileSmartDeleter.hpp